### ISSUE TITLE
[bug] Incorrect message serialization with multiple nested repeated fields

### ISSUE THREAD

**Issue by drygdryg** [PMunch/protobuf-nim#24]
I came across a serialization bug in a more complex case that I mentioned in https://github.com/PMunch/protobuf-nim/issues/23#issuecomment-753461957
I made a simpler example to make it clearer for you:
```protobuf
syntax = "proto3";

message Example {
    message ExampleNested {
        message ExampleNested2 {
            string field1 = 1;
            message ExampleNested3 {
                string field1 = 1;
            }
            repeated ExampleNested3 field2 = 2;
        }
        ExampleNested2 field1 = 1;
    }
    repeated ExampleNested field1 = 1;
}
```
All the necessary files for testing are here: https://gist.github.com/drygdryg/17a7c693330ccf932989fb9401e7eb92

**Comment by drygdryg** [comment]
When I add a single `ExampleNested3` message, it works correctly. But as soon as I add 2 `ExampleNested3` messages, how the length is calculated with an error (it looks like one byte of length is missing).
Correct serialization (made with Python):
![image](https://user-images.githubusercontent.com/43933400/103456902-9a155380-4d0b-11eb-9217-89cce3eafb51.png)
Incorrect:
![image](https://user-images.githubusercontent.com/43933400/103456910-a7324280-4d0b-11eb-822e-c303d9e1217d.png)

**Comment by PMunch** [comment]
The fix for #35 fixed this as well (I think it was the same bug). In any case I can verify that this also produces the correct output now.
